pkg/buildpipeline/pipeline.go

```diff
@@ -730,39 +769,36 @@ func (j *Jenkinsfile) GenerateCRDs(pipelineIdentifier string, buildIdentifier st
 		if len(tasks) == 0 {
 			wsPath = "workspace"
 		}
-		block, err := stageToTask(s, pipelineIdentifier, buildIdentifier, namespace, wsPath, baseEnv, j.Agent, suffix, 0)
+		block, err := stageToTask(s, pipelineIdentifier, buildIdentifier, namespace, wsPath, baseEnv, j.Agent, suffix, 0, nil, previousBlockStage)
 		if err != nil {
 			return nil, nil, err
 		}
+		previousBlockStage = block
 
 		tasks = append(tasks, block.getLinearTasks()...)
-		p.Spec.Tasks = append(p.Spec.Tasks, createPipelineTasks(block, blocks, &blocks)...)
+		p.Spec.Tasks = append(p.Spec.Tasks, createPipelineTasks(block)...)
 	}
 
 	return p, tasks, nil
 }
 
-// Yikes this is a nasty method, would be nice to not need `head`, but without it idk how to handle nested blocks. Needs thought
```

Comment:
No more weird temp blocks or appending to arrays, so I think this is much simpler now.